src/Client/HandlerDelegates.cs

```diff
@@ -0,0 +1,89 @@
+﻿using OmniSharp.Extensions.LanguageServer.Models;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OmniSharp.Extensions.LanguageServerProtocol.Client
+{
+    /// <summary>
+    ///     A handler for empty notifications.
+    /// </summary>
+    /// <returns>
+    ///     A <see cref="Task"/> representing the operation.
+    /// </returns>
+    public delegate void EmptyNotificationHandler();
```

Comment:
Should be safe to rename to `NotificationHandler`.

Reply:
Yeah, I was trying to make it explicit to consumers that empty notifications are not the same as notifications with payloads (the language server implementation will freak out if you don't pass a payload when one is expected.

But I agree it makes more sense to just vary by type parameters.

Reply:
I've changed this, but I'd like to keep the underlying `TryHandleEmptyNotification` methods where there are no generic type parameters because I got caught more than once by accidentally calling the wrong overload.